### ISSUE TITLE
[red-knot] Redudant-cast false positive for `@Todo` types

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ruff#17164]
We have existing handling for filtering out `@Todo` types, but apparently it doesn't work in this case:
```py
from typing import cast, Any

def f(a: Any):
    cast(dict[str, Any], a)  # Value is already of type `@Todo`
```
(https://playknot.ruff.rs/d1038da7-7198-4650-8288-548a94b74c77)
